forge-game/src/main/java/forge/game/ability/effects/MeldEffect.java

```diff
@@ -47,12 +47,19 @@ public void resolve(SpellAbility sa) {
         }
 
         for (Card c : exiled) {
+        boolean attacking = sa.hasParam("Attacking");
             if (c.isToken() || c.getCloneOrigin() != null) {
                 // Neither of these things
                 return;
             } else if (!c.isInZone(ZoneType.Exile)) {
                 return;
             }
+            if (sa.hasParam("EntersTapped")) {
+                c.setTapped(true);
+            }
+            if (sa.hasParam("Attacking")) {
```

Comment:
Do this at the **bottom**:
`        if (addToCombat(movedCard, movedCard.getController(), sa, "Attacking", "Blocking")) {
            game.updateCombatForView();
            game.fireEvent(new GameEventCombatChanged());
        }`

movedCard is what `changeZone()` returns